custom_components/ttlock/sensor.py

```diff
@@ -31,6 +31,7 @@ async def async_setup_entry(
                 LockBattery(coordinator),
                 LockOperator(coordinator),
                 LockTrigger(coordinator),
+                SensorBattery(coordinator) if coordinator.data.sensor else None,
```

Comment:
I'm unsure why mypy isn't flagging this, but async_add_entities appears to take `Iterable[Entity]` and the else results in the type being `Iterable[Entity|None]`

I think the way to make this work is to add an `if entity is not None` to the comprehension like so:

```
async_add_entities(
  [
    entity
    for coordinator ...
    for entity in (...)
    if entity is not None
  ]
)
```